### ISSUE TITLE
Possibillity of adding `chunks` and friends to this crate?

### ISSUE THREAD

**Issue by JSorngard** [cuviper/rayon-cond#5]
Do you believe it would be possible to add the functions

| Serial | Parallel |  
|---|---|  
| `chunks` | `par_chunks` |  
| `chunks_mut` | `par_chunks_mut` |  
| `chunks_mut_exact` | `par_chunks_mut_exact` |  

to this crate? I suspect there might be complications from the fact that these functions are defined on vectors and not iterators, but if you believe it is possible I could take a look at trying to do it, if you do not want to.

**Comment by cuviper** [comment]
> I suspect there might be complications from the fact that these functions are defined on vectors and not iterators,

Right, and so rayon has them in separate `ParallelSlice` and `ParallelSliceMut` extension traits. I guess we could do similar here, but the implementation is simple:

```rust
if parallel {
    CondIterator::Parallel(slice.par_chunks(n))
} else {
    CondIterator::Serial(slice.chunks(n))
}
```
